EventListener/AttachRefreshTokenOnSuccessListener.php

```diff
@@ -107,9 +115,27 @@ public function attachRefreshToken(AuthenticationSuccessEvent $event): void
             $refreshToken = $this->refreshTokenGenerator->createForUserWithTtl($user, $this->ttl);
 
             $this->refreshTokenManager->save($refreshToken);
-            $data[$this->tokenParameterName] = $refreshToken->getRefreshToken();
+            $refreshTokenString = $refreshToken->getRefreshToken();
         }
 
-        $event->setData($data);
+        if ($this->cookie) {
+            $event->getResponse()->headers->setCookie(
+                new Cookie(
+                    $this->tokenParameterName,
+                    $refreshTokenString,
+                    time() + $this->ttl,
+                    $this->cookie['path'],
```

Comment:
Since there aren't any defaults set right now, either defaults need to be set (as suggested above) or checks to make sure all of these keys are defined need to be added with appropriate fallbacks.